src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Queries/Types/TypedContentTypeBuilder.cs

```diff
@@ -62,7 +58,7 @@ public void Build(FieldType contentQuery, ContentTypeDefinition contentTypeDefin
                     .GetOrAdd(part.PartDefinition.Name,
                         partName =>
                         {
-                            queryObjectGraphTypes ??= serviceProvider.GetServices<IObjectGraphType>();
+                            var queryObjectGraphTypes = serviceProvider.GetServices<IObjectGraphType>();
```

Comment:
I was wrong with the service provider, I thought the lambda was long-lived. But the change you made might now trigger multiple resolutions of the same thing. It was ok as the variable was in the method.

Reply:
I don't think it trigger multiple times. The types are registered scooped and only created once when the app starts and the first request is made to the GraphiQL middleware

Reply:
@sebastienros 
At the same time, I do not see a good reason to resolve them in the callback. Why can't we resolve them early out out side the loop? like this

```
            var queryObjectGraphTypes = serviceProvider.GetServices<IObjectGraphType>();
            var queryInputGraphTypes = serviceProvider.GetServices<IInputObjectGraphType>();
```